DEVELOPER_GUIDE.md

```diff
@@ -292,3 +302,28 @@ original PR with an appropriate label `backport <backport-branch-name>` is merge
 run successfully on the PR. For example, if a PR on main needs to be backported to `2.x` branch, add a label
 `backport 2.x` to the PR and make sure the backport workflow runs on the PR along with other checks. Once this PR is
 merged to main, the workflow will create a backport PR to the `2.x` branch.
+
+## Building on Lucene Version updates
+There may be a Lucene version update that can affect your workflow causing errors like
+`java.lang.NoClassDefFoundError: org/apache/lucene/codecs/lucene99/Lucene99Codec`.
+This results in having issues with not being able to do `./gradlew run` or `./gradlew build`.
+
+You can check this [PR](https://github.com/opensearch-project/k-NN/pull/2195) as an example of this event happening
```

Comment:
Can we have a reference on error in neural search rather then in k-NN, can be this one or similar https://github.com/opensearch-project/neural-search/actions/runs/11264856879/job/31381927500#step:4:411

Reply:
Understood, I'll reference a different PR simply because the GH action logs expire after sometime Im going to reference this PR that went through a similar issue
https://github.com/opensearch-project/neural-search/pull/913#issuecomment-2400189329